Basic/Pod/PP.pod

```diff
@@ -25,169 +25,28 @@ PDL::PP - Generate PDL routines from concise descriptions
-function), you need to bear in mind that as each one is generated, all the
-C<pp_addhdr> so far will be included. Therefore, if you add C functions,
-make sure to make them C<static> to avoid clashes with later F<.c> files.
-But a better practice is make them be separate C files, with any necessary
-F<.h> to be included by them and the F<.pd> file. You can then add them
-to your F<Makefile.PL> (note this is the C<_int> version, see separate
-notes on how to "opt-in" for your own modules):
-
-  my @pack = (["pnm.pd", qw(Pnm PDL::IO::Pnm)]);
-  my %hash = pdlpp_stdargs_int(@pack);
-  $hash{OBJECT} .= ' get$(OBJ_EXT)';
-  sub MY::postamble { pdlpp_postamble_int(@pack); }
-  WriteMakefile(%hash);
-
-=head2 pp_addpm
-
-=for ref
-
-Add code to the generated .pm file
-
-=head2 pp_addxs
-
-=for ref
-
-Add extra XS code to the generated XS file
-
-=head2 pp_add_macros
-
-=for ref
-
-Add extra C<$MACRO()> definitions for these functions. Note these generate
-C code. As of 2.080, they will be passed the list of arguments they were
-called with, rather than a single string, split like the C pre-processor
-on commas except if in C<""> or C<()>, with leading and trailing
-whitespace removed.
-
-=for example
-
-  pp_add_macros(SUCC => sub { "($_[0] + 1)" });
-  # ...
-    Code => '$a() = $SUCC($b());',
-
-=head2 pp_add_typemaps
-
-=for ref
-
-Available from 2.082. Add an XS typemap for use as C<OtherPars> or from
-manually-added XS. Takes
-one named argument, either C<typemap> (an L<ExtUtils::Typemaps> object),
-C<string>, or C<file>.
-
-=for example
-
-  pp_add_typemaps(string=><<'EOT');
-  TYPEMAP
-  NV_ADD1 T_NV_ADD1
-
-  INPUT
-  T_NV_ADD1
-    $var = SvNV($arg) + 1;
-
-  OUTPUT
-  T_NV_ADD1
-    sv_setnv($arg, $var - 1);
-  EOT
-  # ...
-    OtherPars => '[o] NV_ADD1 v1',
-
-=head2 pp_beginwrap
-
-=for ref
-
-Add BEGIN-block wrapping to code for the generated .pm file
-
-
-=head2 pp_bless
-
-=for ref
-
-Sets the package to which the XS code is added (default is PDL)
-
-
-=head2 pp_core_importList
-
-=for ref
-
-Specify what is imported from PDL::Core
-
-
-=head2 pp_def
-
-=for ref
-
-Define a new PDL function
-
-
-=head2 pp_deprecate_module
-
-=for ref
-
-Add runtime and POD warnings about a module being deprecated
-
-
-=head2 pp_done
-
-=for ref
-
-Mark the end of PDL::PP definitions in the file
-
-
-=head2 pp_export_nothing
+=head1 OVERVIEW
 
-=for ref
+PDL::PP prepares Perl modules and the C sources ald allows to write
+software which can be called from Perl but executes with C speed -
+with ease.  These C sources need to be compiled before your code can be
+executed.  There are two modes of operation for this:
 
-Clear out the export list for your generated module
+=over
 
+=item Use Inline::Pdlpp
 
-=head2 pp_line_numbers
+With L<Inline::Pdlpp>, the C code will be compiled on the fly at
+runtime.  It is easier to get started with, but you pay the
+compilation overhead for every run.
```

Comment:
If you check the docs of I:Pdlpp (and try it), you'll see it only recompiles when the code changes. Please fix this doc :-)

Also you need to mention it's very hard to make these modules installable, whereas it's very easy with "full" PP modules.